src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,67 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints(dist.entry_points)
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    @staticmethod
+    def _parse_entrypoint(entrypoint: str) -> Tuple[str, str]:
+        """
+        Extract the entry point and name from the string.
+
+        .. code-block:: python
+
+            WritePyproject._parse_entrypoint("hello-world = timmins:hello_world")
+            # returns ('hello-world', 'timmins:hello_world')
+
+            WritePyproject._parse_entrypoint("hello-world")
+            # throws ValueError: Entry point 'hello-world' is not of the form 'name = target'
+
+        :param: entrypoint  The entry point string, of the form
+                            "entrypoint = module:function" (whitespace optional)
+        :returns:           A two-element `tuple`, first element is the entry point name, second element is the target
+                            (module and function name) as a string.
+        :raises ValueError: An equals (`=`) character was not present in the entry point string.
+        """
+        # Format: 'hello-world = timmins:hello_world'
+        if "=" not in entrypoint:
+            raise ValueError("Entry point %r is not of the form 'name = module:function'" % entrypoint)
+
+        (name, target) = entrypoint.split("=", 1)
+        return (name.strip(), target.strip())
+
+    @classmethod
+    def _generate_entrypoints(cls, entrypoints: Optional[Dict[str, List[str]]]) -> Dict[str, Dict[str, str]]:
```

Comment:
This is definitely not the most important thing, but I do think a static method makes more sense here. I mean, there's no need for this method to have access to the class except as a way to access `_parse_entrypoint`, but `_parse_entrypoint` is hard-coded in precisely one class whose name is known and can be referenced directly. A class method would make more sense to me if inheritance were in play, i.e. if there were a hierarchy of different classes that could be passed as `cls`, and this method needed to access potentially different implementations based on what `cls` is, but that's not the case here; `WritePyproject` is serving as nothing more than a container for some related functionality, and the two static methods have no dependence on that class whatsoever.

In fact it could be argued that it would make more sense to put the two methods as module-level functions. I thought about suggesting that in my initial review, but I didn't do so because the methods are never used outside the class and I thought it made sense to minimize the scope in which they're visible, just to make the code easier to follow. (But this is definitely debatable; perhaps they _do_ make more sense as module-level functions.)

Reply:
Yeah, I was avoiding hard-coding the class, but you are right, that's the only reason why it's a class method.

If we made them module-level functions, we can still unit test them individually: `__all__` doesn't stop us importing them and calling them directly (it only specifies what `*` gives you if you do a `from ${module} import *`).

Probably less clutter for the class though, and less probability that we'll confuse people later.

Reply:
Funnily enough, I actually encountered a couple of cases at work this week where I got confused about the purpose of a function that was at top-level scope even though it was only used in a specific local scope. So that's got me leaning further toward a static method being the right solution for these - it keeps them closer to where they're actually used. But I don't think it's worth the trouble to change back at this point; I'm just making this point as something to keep in the back of our mind in the future. If nothing else, it is annoyingly inconsistent to have some helper functions as static methods and others not, so some kind of refactoring will be necessary at some point.

Maybe it will even make sense to break out all the functionality related to entry points into a separate class or even a separate module or something :shrug: but again, not something to deal with now.